state/bakerystorage/config.go

```diff
@@ -0,0 +1,188 @@
+	if err != nil {
+		return txn.Op{}, errors.Trace(err)
+	}
+	externalUsersThirdPartyKeyBytes, err := json.Marshal(externalUsersThirdPartyKey)
+	if err != nil {
+		return txn.Op{}, errors.Trace(err)
+	}
+
+	localUsersKey, err := bakery.GenerateKey()
+	if err != nil {
+		return txn.Op{}, errors.Trace(err)
+	}
+	localUsersKeyBytes, err := json.Marshal(localUsersKey)
+	if err != nil {
+		return txn.Op{}, errors.Trace(err)
+	}
+
+	offersThirdPartyKey, err := bakery.GenerateKey()
+	if err != nil {
+		return txn.Op{}, errors.Trace(err)
+	}
+	offersThirdPartyKeyBytes, err := json.Marshal(offersThirdPartyKey)
+	if err != nil {
+		return txn.Op{}, errors.Trace(err)
+	}
+
+	return txn.Op{
+		C:      b.collection,
+		Id:     bakeryConfigKey,
+		Assert: txn.DocMissing,
+		Insert: &bakeryConfigDoc{
+			LocalUsersKey:              string(localUsersKeyBytes),
+			LocalUsersThirdPartyKey:    string(localUsersThirdPartyKeyBytes),
+			ExternalUsersThirdPartyKey: string(externalUsersThirdPartyKeyBytes),
+			OffersThirdPartyKey:        string(offersThirdPartyKeyBytes),
+		},
+	}, nil
+}
+
+// GetLocalUsersKey returns the key pair used with the local users bakery.
+func (b *bakeryConfig) GetLocalUsersKey() (*bakery.KeyPair, error) {
+	doc, err := b.bakeryConfigDoc()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if doc.LocalUsersKey == "" {
+		return nil, errors.NotValidf("empty local users key")
+	}
+
+	var keyPair bakery.KeyPair
+	err = json.Unmarshal([]byte(doc.LocalUsersKey), &keyPair)
+	if err != nil {
+		err = errors.Trace(err)
+	}
+	return &keyPair, nil
+}
+
+// GetLocalUsersThirdPartyKey returns the third party key pair used with the local users bakery.
+func (b *bakeryConfig) GetLocalUsersThirdPartyKey() (*bakery.KeyPair, error) {
+	doc, err := b.bakeryConfigDoc()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if doc.LocalUsersKey == "" {
+		return nil, errors.NotValidf("empty local users third party key")
+	}
+
+	var keyPair bakery.KeyPair
+	err = json.Unmarshal([]byte(doc.LocalUsersThirdPartyKey), &keyPair)
+	if err != nil {
+		err = errors.Trace(err)
+	}
+	return &keyPair, nil
+}
+
+// GetExternalUsersThirdPartyKey returns the third party key pair used with the external users bakery.
+func (b *bakeryConfig) GetExternalUsersThirdPartyKey() (*bakery.KeyPair, error) {
+	doc, err := b.bakeryConfigDoc()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if doc.LocalUsersKey == "" {
+		return nil, errors.NotValidf("empty external users third party key")
+	}
+
+	var keyPair bakery.KeyPair
+	err = json.Unmarshal([]byte(doc.ExternalUsersThirdPartyKey), &keyPair)
+	if err != nil {
+		err = errors.Trace(err)
+	}
+	return &keyPair, nil
+}
+
+// GetOffersThirdPartyKey returns the key pair used with the cross model offers bakery.
+func (b *bakeryConfig) GetOffersThirdPartyKey() (*bakery.KeyPair, error) {
+	doc, err := b.bakeryConfigDoc()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if doc.LocalUsersKey == "" {
```

Comment:
Perhaps there should be a helper method that takes a json serialized string and a name and returns a key pair and error?

Reply:
good idea